lib/Service/SmimeService.php

```diff
@@ -521,4 +555,45 @@ public function isEncrypted(Horde_Imap_Client_Data_Fetch $message): bool {
 			&& isset($contentType['smime-type'])
 			&& $contentType['smime-type'] === 'enveloped-data';
 	}
+
+	/**
+	 * Encrypt a MIME part using the given certificates.
+	 *
+	 * @param Horde_Mime_Part $part
+	 * @param SmimeCertificate[] $certificates
+	 * @return Horde_Mime_Part New MIME part containing the encrypted message and the signature
+	 *
+	 * @throws SmimeSignException If encrypting the message fails
+	 * @throws ServiceException If decrypting the certificates fails
+	 */
+	public function encryptMimePart(Horde_Mime_Part  $part, array $certificates): Horde_Mime_Part {
+		try {
+			/** @var string[] $decryptedCertificates */
+			$decryptedCertificates = array_map(function (SmimeCertificate $certificate) {
+				return $this->crypto->decrypt($certificate->getCertificate());
+			}, $certificates);
+		} catch (Exception $e) {
+			throw new ServiceException('Failed to decrypt certificate: ' . $e->getMessage(), 0, $e);
+		}
+
+		$inPath = $this->tempManager->getTemporaryFile();
+		$outPath = $this->tempManager->getTemporaryFile();
+		file_put_contents($inPath, $part->toString([
+			'canonical' => true,
+			'headers' => true,
+		]));
+		if (!openssl_pkcs7_encrypt($inPath, $outPath, $decryptedCertificates, [], 0, OPENSSL_CIPHER_AES_128_CBC)) {
```

Comment:
It probably fails because the RC2-* algorithm was deprecated in OpenSSL due to it being unsafe. It's a good idea to explicitly default to a secure and compatible alternative.